docs/pages/machine-id/guides/circleci.mdx

```diff
@@ -0,0 +1,226 @@
+</Details>
+
+In this guide, you will use Teleport Machine ID to allow a CircleCI job to
+securely connect to a Teleport SSH node without the need for long-lived secrets.
+
+This mitigates the risk of long-lived secrets such as passwords or SSH private
+keys being exfiltrated from your CircleCI organization and provides many of
+the other benefits of Teleport such as auditing and finely-grained access
+control.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+(!docs/pages/includes/tctl.mdx!)
+
+- A node that is a part of the Teleport cluster with
+[Server Access](https://goteleport.com/docs/server-access/introduction/) and a
+user you want to grant access to. In this guide, we will use `my-node` and
+`ci-user` respectively, replace these with values appropriate to your setup.
+- A CircleCI project connected to a Git repository you can push to.
+
+## Step 1/4. Configure CircleCI
+
+In order to configure the rules for which CircleCI workflows will be allowed to
+connect to your Teleport cluster, you must determine the ID of your CircleCI
+organization and create a CircleCI context.
+
+### Find your Organization ID
+
+Open CircleCI and navigate to "Organization settings" from the navbar.
+You should be presented with an interface titled "Overview" with a section
+called "Organization ID". Note this value down and substitute `$ORGANIZATION_ID`
+in configuration examples with this.
+
+### Create a Context
+
+CircleCI has an organization level concept called
+["contexts"](https://circleci.com/docs/contexts/). Contexts allow an
+organization to configure a series of secrets that should be exposed to a
+workflow job, and can control which actors are allowed to trigger jobs
+associated with a context.
+
+The contexts that a workflow job has been assigned are also encoded in the
+identity token that CircleCI creates for the job. This makes them an ideal way
+of Teleport determining which CircleCI jobs should be granted access to the
+Teleport cluster.
+
+In this example, you will create a CircleCI context named `teleport-access`.
+You will then grant this context access to your Teleport cluster.
+
+To create the CircleCI context, open up "Organization settings" in CircleCI
+and navigate to "Contexts". Click "Create Context" and provide `teleport-access`
+as the name of the context you wish to create. You may substitute this value
+for a string that makes more sense to your organization, but ensure in future
+steps of this guide that you replace `teleport-access` with your value.
+
+Select the context you have just created, you will now be on a page that allows
+you to configure the context. You will need to determine the ID of the context
+to use when configuring Teleport. This value can be found in the URL of the
+context settings page.
+
+E.G:
+`https://app.circleci.com/settings/organization/github/gravitational/contexts/00000000-0000-0000-0000-000000000000`
```

Comment:
````suggestion
```text
https://app.circleci.com/settings/organization/github/gravitational/contexts/00000000-0000-0000-0000-000000000000
```
````
Suggested to block this off from the surrounding text a bit more